receiver/splunksearchapireceiver/receiver.go

```diff
@@ -111,6 +111,9 @@ func (ssapir *splunksearchapireceiver) Shutdown(ctx context.Context) error {
 
 func (ssapir *splunksearchapireceiver) runQueries(ctx context.Context) error {
```

Comment:
Unless I'm missing something (which I may be. I haven't looked behind the failing test and this function), I believe your test failures are related to this function.

It looks like the logic here is to make your query, wait for it, and then start retrieving the search results and processing them in a loop until the # of search results are less than the max possible returned in a single page. If that's the case you quit the infinite loop, otherwise you try to get another page of results.

Now what happens if it was your last page of results, but the number of results was the max number that can fit within a page (as is with the failing test)? It seems like you try to get search results but get an error because there are none. 

Hopefully I'm right or at least half right and this puts you on the right track to fixing the CI issues.